src/alfasim_score/converter/alfacase/production_operation.py

```diff
@@ -0,0 +1,223 @@
+from typing import Dict
+
+import attr
+from alfasim_sdk import AnnulusEquipmentDescription
+from alfasim_sdk import CaseDescription
+from alfasim_sdk import GasLiftValveEquipmentDescription
+from alfasim_sdk import InitialConditionsDescription
+from alfasim_sdk import InitialPressuresDescription
+from alfasim_sdk import InitialTemperaturesDescription
+from alfasim_sdk import InitialVelocitiesDescription
+from alfasim_sdk import InitialVolumeFractionsDescription
+from alfasim_sdk import MassInflowSplitType
+from alfasim_sdk import MassSourceNodePropertiesDescription
+from alfasim_sdk import MassSourceType
+from alfasim_sdk import MultiInputType
+from alfasim_sdk import PressureContainerDescription
+from alfasim_sdk import PressureNodePropertiesDescription
+from alfasim_sdk import PvtModelCorrelationDescription
+from alfasim_sdk import PvtModelsDescription
+from alfasim_sdk import TableInputType
+from alfasim_sdk import TemperaturesContainerDescription
+from alfasim_sdk import ValveType
+from alfasim_sdk import VelocitiesContainerDescription
+from alfasim_sdk import VolumeFractionsContainerDescription
+from alfasim_sdk._internal.constants import FLUID_GAS
+from alfasim_sdk._internal.constants import FLUID_OIL
+from alfasim_sdk._internal.constants import FLUID_WATER
+from barril.units import Array
+from barril.units import Scalar
+from pathlib import Path
+
+from alfasim_score.common import LiftMethod
+from alfasim_score.common import OperationType
+from alfasim_score.common import convert_api_gravity_to_oil_density
+from alfasim_score.common import convert_gas_gravity_to_gas_density
+from alfasim_score.constants import CO2_MOLAR_FRACTION_DEFAULT
+from alfasim_score.constants import GAS_LIFT_MASS_NODE_NAME
+from alfasim_score.constants import GAS_LIFT_VALVE_DEFAULT_DELTA_P_MIN
+from alfasim_score.constants import GAS_LIFT_VALVE_DEFAULT_DIAMETER
+from alfasim_score.constants import GAS_LIFT_VALVE_DEFAULT_DISCHARGE
+from alfasim_score.constants import GAS_LIFT_VALVE_NAME
+from alfasim_score.constants import H2S_MOLAR_FRACTION_DEFAULT
+from alfasim_score.constants import NULL_VOLUMETRIC_FLOW_RATE
+from alfasim_score.constants import WELLBORE_BOTTOM_NODE_NAME
+from alfasim_score.constants import WELLBORE_TOP_NODE_NAME
+from alfasim_score.converter.alfacase.base_operation import BaseOperationBuilder
+from alfasim_score.units import FRACTION_UNIT
+from alfasim_score.units import LENGTH_UNIT
+from alfasim_score.units import PRESSURE_UNIT
+from alfasim_score.units import TEMPERATURE_UNIT
+from alfasim_score.units import VELOCITY_UNIT
+
+
+class ProductionOperationBuilder(BaseOperationBuilder):
+    def __init__(self, score_filepath: Path):
+        super().__init__(score_filepath)
+        self.operation_type = OperationType.PRODUCTION
+        score_operation_type = self.score_input.read_operation_type()
```

Comment:
Actually, it should be used to define which builder will be used.

Reply:
I'm still thinking which is the better strategy to do this selection. Maybe a function to chose which class should be instantiated, maybe a if block in the main converter function... (other one?).

Reply:
Think as a user. For example, I'd like to just call it in my code:

```
base_path = "/score/projects/well_1/"
score_input_file = "study_01.json"
alfacase_file = "study_01.alfacase"
alfacase_converter = ScoreAlfacaseConverter(base_path  + score_input_file )
alfacase_converter.build_alfacase(base_path  + alfacase_file )
```

All handle to create `ScoreInputReader`, `ProductionOperationBuilder` or `InjectionOperationBuilder` should be performed in `ScoreAlfacaseConverter`.


You actually dit it for each operation, thus currently what we have is:

```
base_path = "/score/projects/well_1/"
score_input_file = "study_01.json"
alfacase_file = "study_01.alfacase"

operation_type = "PRODUCTION"

builder = None

if operation_type  is "PRODUCTION":
    builder  = ProductionOperationBuilder(score_input_file )
elif operation_type  is "INJECTION":
    builder  = InjectionOperationBuilder(score_input_file )
else:
    raise TypeError("Only PRODUCTION and INJECTION is allowed as operation")

case_description = builder.generate_operation_alfacase_description()
alfacase = convert_description_to_alfacase(case_description)

alfacase_path = base_path  + alfacase_file
alfacase_path.write_text(alfacase, encoding='utf-8')
```

is it correct?


Let'st talk about it. It is not a issue for this PR. But we can discuss it for an improvement if necessary.